ignite/metrics/loss.py

```diff
@@ -30,9 +30,52 @@ class Loss(Metric):
             metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
             non-blocking. By default, CPU.
 
+    Attributes:
+        required_output_keys: dictionary defines required keys to be found in ``engine.state.output`` if the
+            latter is a dictionary. Default, ``("y_pred", "y", "criterion_kwargs")``. This is useful when the
+            criterion function requires additional argument, which can be passed using ``criterion_kwargs``.
+            See notes below for an example.
+
+    Note:
+
+        Let's implement a Loss metric that requires ``y_pred``, ``y`` and ``criterion_kwargs`` as input
+        for ``criterion`` function. In the example below we show how to setup standard metric like Accuracy
+        and the Loss metric using an ``evaluator`` created with
+        :meth:`~ignite.engine.create_supervised_evaluator` method.
+
+        .. code-block:: python
+
+            import torch
+            import torch.nn as nn
+            from torch.nn.functional import nll_loss
+
+            from ignite.metrics import Accuracy, Loss
+            from ignite.engine import create_supervised_evaluator
+
+            model = ...
+
+            criterion = nll_loss
+
+            metrics = {
+                "Accuracy": Accuracy(),
+                "Loss": Loss(criterion)
+            }
+
+            # global criterion kwargs
+            criterion_kwargs = {...}
+
+            evaluator = create_supervised_evaluator(
+                model,
+                metrics=metrics,
+                output_transform=lambda x, y, y_pred: {
+                    "x": x, "y": y, "y_pred": y_pred, "criterion_kwargs": criterion_kwargs}
```

Comment:
Here it isn't clear where the `x` is coming from since there is no mention of this variable in the above lines. Maybe we can include it in line 41 or just exclude this variable altogether?

Reply:
This is the signature for create_supervised_evaluator of evaluate: 
https://github.com/pytorch/ignite/blob/dce97dfeacf1516e78b1cd9c6a37e57a1228ceed/ignite/engine/__init__.py#L518-L527

It contains lambda with 3 values `(x, y_pred, y)`, and by default, it takes 2 arguments `(y_pred, y)`. I can do something like this; refrain from passing x, but the lambda must contain x, whether it is used or not otherwise it will throw an error, code for this approach:

```python

evaluator = create_supervised_evaluator(
    model,
    metrics=metrics,
    output_transform=lambda x, y, y_pred: {
        "y": y, "y_pred": y_pred, "criterion_kwargs": criterion_kwargs}
)
```

Reply:
Let's not remove `x` then since it won't make sense. We can just add `x` to line 41 like - requires `y_pred`, `y`, `x` and `criterion_kwargs`... 

As previously mentioned [here](https://pytorch.org/ignite/_modules/ignite/metrics/metric.html#Metric).

Reply:
Ok